arangod/Cluster/ClusterInfo.cpp

```diff
@@ -3054,6 +3197,24 @@ Result ClusterInfo::createCollectionsCoordinator(
     opers.emplace_back(CreateCollectionOrder(databaseName, info.collectionID,
                                              info.isBuildingSlice()));
 
+    if (replicationVersion == replication::Version::TWO) {
+      // Create a replicated state for each shard.
+      replicatedStates.reserve(replicatedStates.size() + shardServers.size());
```

Comment:
Why `replicatedStates.size() + shardServers.size()`? Shouldn't this just be `shardServers.size()`?

Reply:
Unfortunately we are inside a huge for loop starting at line 3028, `for (auto& info : infos)`, hence we might end up with multiple `reserve` calls, depending on the number of collections. Not great, but didn't think of something else at that point.